pkg/controller/test/context_builder.go

```diff
@@ -344,6 +360,9 @@ func (b *Builder) Sync() {
 	if b.additionalSyncFuncs != nil {
 		cache.WaitForCacheSync(b.stopCh, b.additionalSyncFuncs...)
 	}
+
+	// Wait for the informerResyncPeriod to make sure any update made by any of the fake clients
+	// is reflected in the informer caches.
 	time.Sleep(informerResyncPeriod)
```

Comment:
I don't really understand why we are pausing that much here, 1 second seems like very long for that purpose.

Reply:
That is what I was trying to explain in the comments 😅:
Sync is not only used on startup, it is also used after performing an operation, eg. Update() -> Sync() -> Get()
I'll update the comments to better explain this.

Reply:
Updated.

Reply:
I think I understand this now thanks to this new comment.
In another PR it would be great to add more content to the `doc.go` file and maybe an example or two,
explaining how this package is intended to be used....something that will make sense when viewed by contributors at:
 * https://pkg.go.dev/github.com/cert-manager/cert-manager@v1.13.3/pkg/controller/test